### PR TITLE
Octane fixes 5.8.7 

### PR DESCRIPTION
**Pull request by radislavB**
[JENKINS-59170 : UFT executed in workflow job - is not recognized as UFT test in ALM Octane](https://issues.jenkins-ci.org/browse/JENKINS-59170)
[JENKINS-59174 : UFT executor job created without UFT slave label , because of existence of non-letter characters in slave name](https://issues.jenkins-ci.org/browse/JENKINS-59174)

### CODE REVIEW

File: src/main/java/com/microfocus/application/automation/tools/octane/tests/junit/JUnitExtension.java [hunk 1]
```diff
@@ -150,13 +152,17 @@ private ResultFields getResultFields(Run<?, ?> build, HPRunnerType hpRunnerType,
...
+	private boolean isLoadRunnerProject(Run run) {
+		try {
+			FilePath performanceReportFolder = new FilePath(run.getRootDir()).child(PERFORMANCE_REPORT);
+			FilePath transactionSummaryFolder = new FilePath(run.getRootDir()).child(TRANSACTION_SUMMARY);
+			return performanceReportFolder.exists() &&
+					performanceReportFolder.isDirectory() &&
+					transactionSummaryFolder.exists() &&
+					transactionSummaryFolder.isDirectory();
+		} catch (IOException | InterruptedException e) {
+			return false;
```
Comment (NarcisaMGalan) [inline src/main/java/com/microfocus/application/automation/tools/octane/tests/junit/JUnitExtension.java]: Shouldn't you use logger here?

File: src/main/java/com/microfocus/application/automation/tools/octane/tests/TestListener.java [hunk 2]
```diff
@@ -56,24 +60,34 @@ public boolean processBuild(Run run) {
...
-				}
-				if (STORMRUNNER_FUNCTIONAL_TEST_RUNNER_CLASS.equals(builder.getClass().getSimpleName())) {
-					hpRunnerType = HPRunnerType.StormRunnerFunctional;
-					break;
+		if (JobProcessorFactory.WORKFLOW_RUN_NAME.equals(run.getClass().getName())) {
+			ParametersAction parameterAction = run.getAction(ParametersAction.class);
+			if (parameterAction != null) {
+				ParameterValue pv = parameterAction.getParameter(HPRunnerType.class.getSimpleName());
+				if (pv != null) {
+					hpRunnerType = HPRunnerType.valueOf((String) pv.getValue());
```
Comment (NarcisaMGalan) [inline src/main/java/com/microfocus/application/automation/tools/octane/tests/TestListener.java]: Please try to change here, there are too many if statements?

File: src/main/java/com/microfocus/application/automation/tools/octane/tests/TestListener.java [hunk 3]
```diff
@@ -56,24 +60,34 @@ public boolean processBuild(Run run) {
...
+						break;
+					}
+					if (RUN_FROM_FILE_BUILDER.equals(builder.getClass().getSimpleName()) || RUN_FROM_ALM_BUILDER.equals(builder.getClass().getSimpleName())) {
+						hpRunnerType = HPRunnerType.UFT;
+						break;
+					}
+					if (PERFORMANCE_CENTER_TEST_RUNNER_CLASS.equals(builder.getClass().getSimpleName())) {
+						hpRunnerType = HPRunnerType.PerformanceCenter;
+						break;
+					}
```
Comment (NarcisaMGalan) [inline src/main/java/com/microfocus/application/automation/tools/octane/tests/TestListener.java]: Here the same

File: src/test/java/com/microfocus/application/automation/tools/octane/tests/detection/UFTExtensionTest.java [hunk 4]
```diff
@@ -42,6 +42,7 @@
 import java.util.UUID;
 
 @SuppressWarnings("squid:S2699")
+@Ignore
```
Comment (NarcisaMGalan) [inline src/test/java/com/microfocus/application/automation/tools/octane/tests/detection/UFTExtensionTest.java]: Why do you use @Ignore here?

**Comment by radislavB** [comment]
I did refactoring to respond to your request.

**Pull request merged**